core/src/main/scala/com/microsoft/azure/synapse/ml/exploratory/DistributionBalanceMeasure.scala

```diff
@@ -146,6 +179,23 @@ class DistributionBalanceMeasure(override val uid: String)
         Nil
     )
   }
+
+  override def validateSchema(schema: StructType): Unit = {
+    super.validateSchema(schema)
+
+    getReferenceDistribution.isEmpty match {
+      case true =>
+      case false if getReferenceDistribution.length != getSensitiveCols.length => throw new Exception(
+        "The reference distribution must have the same length and order as the sensitive columns " +
+          getSensitiveCols.mkString(", "))
+      case false => getReferenceDistribution.foreach {
+        case _: Map[String, Double] | null => //scalastyle:ignore null
+        case m => throw new Exception(
```

Comment:
Scala is saying this is unreachable code:
![image](https://user-images.githubusercontent.com/64443771/226762858-50659de4-0bda-48d3-ae43-9a5b3de7703d.png)

I added this so that it will throw the exception if a user tries to pass in a non-map in pyspark like this:
```
reference_dists = ["string", 0]
new DistributionBalanceMeasure()
   .setReferenceDistribution(reference_dists)
```

I actually haven't tested this in pyspark yet (didn't get to set up the python conda env and build wheel), so should I worry about this scenario?

Reply:
the code is strongly typed here so it would be an error before this case match statement

Reply:
Sounds good! I removed this case in the latest commit.

Semi-related: I noticed that the `refDist.isEmpty` check was glossing over whether the user actually passed in a custom distribution or not (because we default to empty). So instead, I pushed a change to use `isDefined` and removed the default case.